### PR TITLE
Remove -Wno-unused-but-set-variable suppression for GCC

### PR DESCRIPTION
**Pull request by spham-amzn**
- Remove -Wno-unused-but-set-variable from the GCC compilation options
- Fix resulting errors caused by -Wunused-but-set-variable

Signed-off-by: Steve Pham <82231385+spham-amzn@users.noreply.github.com>

### CODE REVIEW

**Review by AMZN-ScottR — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
This is awesome! Great work @spham-amzn 😄

File: cmake/Platform/Common/GCC/Configurations_gcc.cmake [hunk 1]
```diff
@@ -72,7 +72,6 @@ ly_append_configurations_options(
         -Wno-stringop-truncation
         -Wno-switch
         -Wno-uninitialized
-        -Wno-unused-but-set-variable
```
Comment (moraaar) [inline cmake/Platform/Common/GCC/Configurations_gcc.cmake:75]: question: This warning has helped to improve the code by removing unused variables (which could be a bug), are we sure we want to remove it? Maybe we can add it to the other compilers so AR picks them up too? thoughts?

**Review by moraaar — APPROVED** [review]
Great clean up. Approved. Asked a question that I'd like to know what you think.

File: cmake/Platform/Common/GCC/Configurations_gcc.cmake (hunk 1 above)
Comment (moraaar) [reply cmake/Platform/Common/GCC/Configurations_gcc.cmake:75]: Silly me, you're right. I misread it. Cheers.

File: Gems/GradientSignal/Code/Source/Platform/Linux/platform_linux_editor.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply Gems/GradientSignal/Code/Source/Platform/Linux/platform_linux_editor.cmake:9]: Maybe I should move this out to cmake/Platforms as well

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**